landoapi/models/landing_job.py

```diff
@@ -82,30 +82,11 @@ class LandingJob(Base):
     # ordered, the resulting column will have the same order
     # as the definition order of the enum. This can be relied
     # on for comparisons or things like queries with ORDER BY.
-    status = db.Column(
-        db.Enum(LandingJobStatus), nullable=False, default=LandingJobStatus.SUBMITTED
-    )
-
-    # JSON object mapping string revision id of the form "<int>" (used because
-    # json keys may not be integers) to integer diff id. This is used to
-    # record the diff id used with each revision and make searching for
-    # Transplants that match a set of revisions easy (such as those
-    # in a stack).
-    # e.g.
-    #     {
-    #         "1001": 1221,
-    #         "1002": 1246,
-    #         "1003": 1412
-    #     }
-    revision_to_diff_id = db.Column(JSONB, nullable=False)
-
-    # JSON array of string revision ids of the form "<int>" (used to match
-    # the string type of revision_to_diff_id keys) listing the order
-    # of the revisions in the request from most ancestral to most
-    # descendant.
-    # e.g.
-    #     ["1001", "1002", "1003"]
-    revision_order = db.Column(JSONB, nullable=False)
+    status = db.Column(db.Enum(LandingJobStatus), nullable=True, default=None)
+
+    # revision_to_diff_id and revision_order are deprecated and kept for historical reasons.
+    revision_to_diff_id = db.Column(JSONB, nullable=True)
+    revision_order = db.Column(JSONB, nullable=True)
```

Comment:
Is there going to be special handling for legacy records here? Perhaps a data migration should be made to migrate the old legacy fields to new ones with `Revision` records.